tests/test_core.py

```diff
@@ -7595,21 +7611,33 @@ def setUp(self):
     self.emcc_args = emcc_args[:]
     for k, v in settings.items():
       self.set_setting(k, v)
+
+    # keep a copy of the settings for this test mode, so we can know
+    # them even in tests that modify settings
+    self.test_mode_settings = settings
+
+    # avoid various compiler warnings in our test output
     Building.COMPILER_TEST_OPTS += [
-        '-Werror', '-Wno-dynamic-class-memaccess', '-Wno-format',
-        '-Wno-format-extra-args', '-Wno-format-security',
-        '-Wno-pointer-bool-conversion', '-Wno-unused-volatile-lvalue',
-        '-Wno-c++11-compat-deprecated-writable-strings',
-        '-Wno-invalid-pp-token', '-Wno-shift-negative-value'
+      '-Werror', '-Wno-dynamic-class-memaccess', '-Wno-format',
+      '-Wno-format-extra-args', '-Wno-format-security',
+      '-Wno-pointer-bool-conversion', '-Wno-unused-volatile-lvalue',
+      '-Wno-c++11-compat-deprecated-writable-strings',
+      '-Wno-invalid-pp-token', '-Wno-shift-negative-value'
     ]
 
+    # forward important flags to COMPILER_TEST_OPTS, which is sent via
+    # CFLAGS etc. to libraries that are built.
     for arg in self.emcc_args:
+      # propagate optimization level
       if arg.startswith('-O'):
-        Building.COMPILER_TEST_OPTS.append(arg) # so bitcode is optimized too, this is for cpp to ll
+        Building.COMPILER_TEST_OPTS.append(arg)
       # propagate LLVM machine arguments
       if arg.startswith('-m'):
         Building.COMPILER_TEST_OPTS.append(arg)
 
+    for key, value in settings.items():
+      Building.COMPILER_TEST_OPTS += ['-s', '%s=%s' % (key, json.dumps(value))]
```

Comment:
The problem is that this ends up duplicating the settings on the command line.   See `get_emcc_args`.

I think we might need to revisit the use of `would like to see is revise the meaning usage of `COMPILER_TEST_OPTS`

Reply:
Good point, thanks - I'll investigate removing/unifying `COMPILER_TEST_OPTS`.